### PR TITLE
New `vagrant pristine` command

### PR DESCRIPTION
**Pull request by fgrehm**
Closes GH-5378

Ideally we should just reuse the other command plugins instead of copying and pasting but its been a while since I been around the codebase and I can't remember if it would be possible to do that :blush: 

/cc @sethvargo

### CODE REVIEW

File: plugins/commands/pristine/plugin.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+module VagrantPlugins
+  module CommandPristine
+    class Plugin < Vagrant.plugin("2")
+      name "pristine command"
+      description <<-DESC
+      The `pristine` command is a shortcut for running `destroy` and `up`
+      DESC
+
+      command("pristine") do
+        require File.expand_path("../command", __FILE__)
```
Comment (sethvargo) [inline plugins/commands/pristine/plugin.rb]: Let's use `require_relative` instead please

File: plugins/commands/pristine/command.rb [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+          @env.ui.info("", prefix: false)
+
+          m.ui.success(I18n.t(
+            "vagrant.post_up_message",
+            name: m.name.to_s,
+            message: m.config.vm.post_up_message))
+        end
+
+        # Success, exit status 0
+        0
```
Comment (sethvargo) [inline plugins/commands/pristine/command.rb]: Explicit `return` here would be nice :smile:

**Comment by sethvargo** [comment]
Hey @fgrehm 

Thank you for the Pull Request. At a high level, this looks really great. Do you think you could add some tests for this? If you look at the pushes tests, I think you can find a good pattern to use. I'm just really hesitant to merge such a large change without test coverage. Even if it works now, we need to prevent regressions in the future :smile:

**Comment by fgrehm** [comment]
Sure! I'll try to write those tests over the week.

**Comment by mitchellh** [comment]
Thanks for this. I'm not a big fan of the name, I don't know what to expect with `pristine`, for example I still have no idea what `gem pristine` does. I know what "pristine" means but for some reason it has never resonated with me as a command name. I still think this makes sense as a flag to `reload`.

**Comment by mitchellh** [comment]
.@sethvargo is trying to convince me this should be a command and not a flag. We'll delay this for 1.8 in some form.

**Comment by fgrehm** [comment]
No worries! I'm not sure when I'll have a chance to write tests for this so feel free to close the PR if you guys want to hold back on it! :beers:

**Comment by sethvargo** [comment]
Hey @fgrehm 

Thanks for the PR. Coming around back to this (almost a year later), I think I've had a change of heart. After reading through all the comments in the original issue, I think everyone's opinions on "what this should be named" and "what it should do" are too far apart. As such, I'm going to recommend folks use a shell alias to involve whatever their preferred behavior is, instead of building this into core. I totally appreciate your efforts on this PR, but I'm actively working to simplify Vagrant's CLI API, and I don't think it's wise to add new commands at this time, especially since this "works" with some `&&`. Sorry!

**Comment by fgrehm** [comment]
No need to be sorry! 🍻

**Pull request closed without merging**